adapters/Fyber/FyberAdapter/GADMAdapterFyberInitializer.m

```diff
@@ -0,0 +1,100 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterFyberInitializer.h"
+
+#import <IASDKCore/IASDKCore.h>
+
+#import "GADMAdapterFyberConstants.h"
+#import "GADMAdapterFyberUtils.h"
+
+/// Fyber Marketplace SDK init state.
+typedef NS_ENUM(NSInteger, GADMAdapterFyberInitState) {
+  GADMAdapterFyberInitStateUninitialized,  ///< Fyber Marketplace SDK is not initialized.
+  GADMAdapterFyberInitStateInitializing,   ///< Fyber Marketplace SDK is initializing.
+  GADMAdapterFyberInitStateInitialized     ///< Fyber Marketplace SDK is initialized.
+};
+
+@implementation GADMAdapterFyberInitializer {
+  /// Fyber Marketplace SDK init state.
+  GADMAdapterFyberInitState _initState;
+
+  /// An array of Fyber adapter initialization completion handler.
+  NSMutableArray<GADMAdapterFyberInitCompletionHandler> *_completionHandlers;
+}
+
++ (nonnull GADMAdapterFyberInitializer *)sharedInstance {
+  static dispatch_once_t onceToken;
+  static GADMAdapterFyberInitializer *instance;
+  dispatch_once(&onceToken, ^{
+    instance = [[GADMAdapterFyberInitializer alloc] init];
+  });
+  return instance;
+}
+
+- (nonnull instancetype)init {
+  self = [super init];
+  if (self) {
+    _initState = GADMAdapterFyberInitStateUninitialized;
+    _completionHandlers = [[NSMutableArray alloc] init];
+  }
+  return self;
+}
+
+- (void)initializeWithAppId:(nonnull NSString *)appID
+          completionHandler:(nonnull GADMAdapterFyberInitCompletionHandler)completionHandler {
+  if (_initState == GADMAdapterFyberInitStateInitialized) {
+    completionHandler(nil);
+    return;
+  }
+
+  if (!appID.length) {
+    NSError *error = GADMAdapterFyberErrorWithCodeAndDescription(
+        kGADErrorInternalError, @"Missing or invalid Application ID.");
+    completionHandler(error);
+    return;
+  }
+
+  GADMAdapterFyberMutableArrayAddObject(_completionHandlers, completionHandler);
+  if (_initState == GADMAdapterFyberInitStateInitializing) {
+    return;
+  }
+  _initState = GADMAdapterFyberInitStateInitializing;
+
+  GADMAdapterFyberLog(@"Initializing Fyber Marketplace SDK with Application ID: %@.", appID);
+  GADMAdapterFyberInitializer *__weak weakSelf = self;
+  [IASDKCore.sharedInstance
+        initWithAppID:appID
+      completionBlock:^(BOOL success, NSError *_Nullable error) {
+        GADMAdapterFyberInitializer *strongSelf = weakSelf;
+        if (!strongSelf) {
+          return;
+        }
+
+        if (error) {
```

Comment:
@joshuajplagonera did you mean `if (success)`?

Reply:
Fixed. Thank you !